src/compiler/checker.ts

```diff
@@ -1828,14 +1828,19 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     };
 
     function runWithoutResolvedSignatureCaching<T>(node: Node | undefined, fn: () => T): T {
-        const containingCall = findAncestor(node, isCallLikeExpression);
-        const containingCallResolvedSignature = containingCall && getNodeLinks(containingCall).resolvedSignature;
-        if (containingCall) {
-            getNodeLinks(containingCall).resolvedSignature = undefined;
+        const cachedSignatures = [];

```

Comment:
It is important to do this for all signatures on the ancestry path because during completion requests diagnostics are *not* reported, this is reflected by `const reportErrors = !isInferencePartiallyBlocked && !candidatesOutArray;` in `resolveCall`. But if we only restore the result of the inner call (when doing a completion request within its argument) then we might cache the outer signature. That signature might even be correct - but what happens is that when actual diagnostics are requested we don't ever get to checking the inner call because a regular typecheck happens from the top to the bottom and the outer `checkCallExpression` could just use the cached signature and bail out from reaching the inner `checkCallExpression`